### DISCUSSION TITLE
Create rule publishing values from `status`

### DISCUSSION THREAD

**Discussion by guydvir2** [arendst/Tasmota#11295] [Q&A] [answered]
How do I publish a MQTT message with current IP ?
`Rule1 ON POWER1#state=1 DO publish myHome/Messages status5#IpAddress ENDON` is not working ( pressing a button is just a trigger )

Guy

**Comment by sfromis** [comment]
You can't use trigger expressions with # anywhere but in the trigger part of a rule. If you execute the command [IPAddress](https://tasmota.github.io/docs/Commands/#ipaddress) from a rule, Tasmota will publish the result.  Using `Backlog IPAddress` from a rule, you can use a rule like `on IPAddress1 do var1 %value% endon` to capture individual fields.

Tasmota already publishes IP address when it is assigned, after boot or reconnect.
`..../INFO2 = {"WebServerMode":"Admin","Hostname":"xxxxxxxx","IPAddress":"192.168.xxx.xxx"}`

**Reply by guydvir2** [reply]
So when `IPAddress1`'s value changes ( ? at some point during boot? ), it will store its value for later use in a post ?

**Reply by sfromis** [reply]
If I were to use the IP address for anything (not a good strategy IMO), I'd save it from the /INFO2 message received on the subscription. I'd not create **yet another** place for the Tasmota device to republish the IP address. If I wanted to query it from backend, I'd just publish the IPAddress command (alternatively publish the Status 5 command).

**Comment by barbudor** [comment, marked as answer]
It's not possible to do as you are doing
You must use a boot time rule with backlog to capture the output of a command such as
`on system#boot do backlog IpAddress endon on IpAddress#IpAddress1 do var1 %value% endon`
Then you have the IP address in var1 that you can re-use in a `publish myhome/messages %var1%`

It looks like youu are trying to turning around all existing Tasmota mecanisms
Information that you are expecting are already publish at boot time by Tasmota or can be requested by your backend

**Reply by sfromis** [reply]
`on IpAddress#IpAddress1 do var1 %value% endon` won't work ;-) There is no "IpAddress" object in the JSON.

**Reply by guydvir2** [reply]
@barbudor regarding your last sentence, you may be right, since I'm until now I used my own Arduino code, and MQTT concept how to share data ( some time for a group ). 
It is easier for me to "ask" a group, or a single device - what is its IP address, than entering Router's WEB page and seeking for the device.

From what you say `IpAddress` is an actual command, OK - but what I was thinking how to retrieve from JSON's ` status 5`  `IpAddress`  -isn't possible ?

**Reply by sfromis** [reply]
The IP address from Status 5 is the same as the one from the shorter IPAddress payload. You can use a trigger like `on StatusNET#IPAddress do .... endon`

More generally, I simply do not use IP addresses for talking to devices. Primarily I do **all** via MQTT, or I use the configurable host name, which my router enters into DNS lookup. The IP address is incidental, and rarely used for anything.

**Reply by guydvir2** [reply]
@sfromis I also use MQTT, but I still need to now device's IP for WEBUI for example. Due to the fact that somehow device's ip changed somehow ( perhaps it was offline for long enough to lose its lease ), I couldnt reach it's WEBUI. 

Regarding your reply to @barbudor  - your `rule` syntax did not work as well

**Reply by barbudor** [reply]
```
14:09:12.309 CMD: rule1
14:09:12.316 MQT: stat/nodemcu/RULE = {"Rule1":"ON","Once":"OFF","StopOnError":"OFF","Length":44,"Free":467,"Rules":"on statusnet#ipaddress do var1 %value% endon"}
14:09:15.414 CMD: status 5
14:09:15.421 MQT: stat/nodemcu/STATUS5 = {"StatusNET":{"Hostname":"nodemcu","IPAddress":"192.168.168.195","Gateway":"192.168.168.254","Subnetmask":"255.255.255.0","DNSServer":"1.1.1.1","Mac":"18:FE:34:A6:D6:83","Webserver":2,"WifiConfig":5,"WifiPower":17.0}}
14:09:15.444 RUL: STATUSNET#IPADDRESS performs "var1 192.168.168.195"
14:09:15.452 MQT: stat/nodemcu/VAR = {"Var1":"192.168.168.195"}
```

**Reply by barbudor** [reply]
for the other rule, as pointed to sfromis the syntax was not correct
here is the correct one
```
14:12:17.686 MQT: stat/nodemcu/RULE = {"Rule1":"ON","Once":"OFF","StopOnError":"OFF","Length":35,"Free":476,"Rules":"on ipaddress1 do var1 %value% endon"}
14:12:19.749 CMD: ipaddress
14:12:19.757 MQT: stat/nodemcu/IPADDRESS = {"IPAddress1":"192.168.168.195 (192.168.168.195)","IPAddress2":"192.168.168.254","IPAddress3":"255.255.255.0","IPAddress4":"1.1.1.1"}
14:12:19.773 RUL: IPADDRESS1 performs "var1 192.168.168.195 (192.168.168.195)"
14:12:19.781 MQT: stat/nodemcu/VAR = {"Var1":"192.168.168.195 (192.168.168.195"}
```

**Reply by barbudor** [reply]
But also you are trying to redo things that Tasmota alread does natively: announcing all this on MQTT at boot or on respond to MQTT command.
This would be more efficient to configure your backend to Tasmota than the opposite way.
